definitions/procedures/service/list.rb

```diff
@@ -16,12 +16,22 @@ def run
 
     def unit_files_list(services)
       if systemd_installed?
-        regex = services.map { |service| "^#{service.name}.service" }.join('\|')
-        execute("systemctl list-unit-files | grep '#{regex}'")
+        execute("systemctl list-unit-files --type=service | \
+                 grep '#{systemd_service_names_regex(services)}'")
       else
         regex = services.map { |service| "^#{service.name} " }.join('\|')
         execute("chkconfig --list 2>&1 | grep '#{regex}'")
       end
     end
+
+    def systemd_service_names_regex(services)
```

Comment:
```suggestion
    def build_regex_for_services(services)
```
Could you rename this method to something similar and not just specific to `systemd_service`?

Reply:
I named it `systemd_service_names_regex' because only systemd service has 'instance_parent_unit'.